assembl/static2/js/app/components/debate/survey/post.jsx

```diff
@@ -51,31 +53,35 @@ class Post extends React.Component {
   }
   handleAddSentiment(target, type) {
     const { id } = this.props;
-    this.props.addSentiment({ variables: { postId: id, type: type } })
-    .then((sentiments) => {
-      target.setAttribute('class', 'sentiment sentiment-active');
-      this.setState({
-        like: sentiments.data.addSentiment.post.sentimentCounts.like,
-        disagree: sentiments.data.addSentiment.post.sentimentCounts.disagree,
-        mySentiment: sentiments.data.addSentiment.post.mySentiment
+    this.props
+      .addSentiment({ variables: { postId: id, type: type } })
+      .then((sentiments) => {
+        target.setAttribute('class', 'sentiment sentiment-active');
+        this.setState({
+          like: sentiments.data.addSentiment.post.sentimentCounts.like,
+          disagree: sentiments.data.addSentiment.post.sentimentCounts.disagree,
+          mySentiment: sentiments.data.addSentiment.post.mySentiment
+        });
+      })
+      .catch((error) => {
+        displayAlert('danger', `${error}`);
       });
-    }).catch((error) => {
-      displayAlert('danger', `${error}`);
-    });
   }
   handleDeleteSentiment(target) {
     const { id } = this.props;
-    this.props.deleteSentiment({ variables: { postId: id } })
-    .then((sentiments) => {
-      target.setAttribute('class', 'sentiment');
-      this.setState({
-        like: sentiments.data.deleteSentiment.post.sentimentCounts.like,
-        disagree: sentiments.data.deleteSentiment.post.sentimentCounts.disagree,
-        mySentiment: sentiments.data.deleteSentiment.post.mySentiment
+    this.props
+      .deleteSentiment({ variables: { postId: id } })
+      .then((sentiments) => {
+        target.setAttribute('class', 'sentiment');
+        this.setState({
+          like: sentiments.data.deleteSentiment.post.sentimentCounts.like,
+          disagree: sentiments.data.deleteSentiment.post.sentimentCounts.disagree,
+          mySentiment: sentiments.data.deleteSentiment.post.mySentiment
```

Comment:
Oh ouch. @cedricmessiant What do you think, maybe we can make models out of these things, like sentiments, posts, etc etc, so that we can have greater control over the state of these objects when they are live. Of course, what I'm saying requires LOTs of boilerplate to keep models consistent with the Apollo state, future requests, etc, etc. What is your opinion on this?